spec/integration/puppet_x/puppetlabs/powershell_manager_spec.rb

```diff
@@ -165,6 +169,24 @@ class PowerShellManager; end
       expect(first_pid).to eq(second_pid)
     end
 
+    it "should create a new PowerShell manager host if user code exits the first process" do
+      first_pid = manager.execute('[Diagnostics.Process]::GetCurrentProcess().Id')[:stdout]
+      exitcode = manager.execute('[Diagnostics.Process]::GetCurrentProcess().Kill()')[:exitcode]
+
+      # when a process gets torn down out from under manager, there is no response
+      # a nil exitcode is a reasonable signal that the PS process was torn out from under it
+      expect(exitcode).to eq(nil)
```

Comment:
@jpogran / @ferventcoder / @glennsarti / @MosesMendoza - thoughts here on this?

If the underlying PS dies (whether intentionally or unintentionally crashes for instance) - then there will be no `exitcode` at all.  The provider should probably recognize this and behave accordingly at https://github.com/puppetlabs/puppetlabs-powershell/blob/master/lib/puppet/provider/exec/powershell.rb#L81

Any thoughts there?  What do other exec providers do if the process crashes?